src/main/scala/mesosphere/marathon/api/v2/LabelSelector.scala

```diff
@@ -0,0 +1,84 @@
+package mesosphere.marathon.api.v2
+
+import mesosphere.marathon.state.AppDefinition
+
+import scala.util.control.NonFatal
+import scala.util.parsing.combinator.RegexParsers
+
+case class LabelSelector(key: String, fn: String => Boolean, value: List[String]) {
+  def matches(app: AppDefinition): Boolean = app.labels.contains(key) && fn(app.labels(key))
+
+}
+case class LabelSelectors(selectors: Seq[LabelSelector]) {
+  def matches(app: AppDefinition): Boolean = selectors.forall(_.matches(app))
+}
+
+/**
+  * Parse a label selector query.
+  * A label selector query has this format:
+  * Query: Selector {, Selector}
+  * Selector: ExistenceSelector | SetSelector | EqualsSelector
+  * ExistenceSelector: Id
+  * SetSelector: Label in|notin Set
+  * EqualsSelector: Label ==|!= Ident
+  * Set: ( Ident {, Ident} )
+  * Label: any character without whitespace
+  * Ident: any character without , or )
```

Comment:
What about matching fancy values with funny characters? You could also allow string literals.